public/app/plugins/datasource/loki/migrations/variableQueryMigrations.ts

```diff
@@ -0,0 +1,36 @@
+import { LokiVariableQuery, LokiVariableQueryType } from '../types';
+
+export const labelNamesRegex = /^label_names\(\)\s*$/;
+export const labelValuesRegex = /^label_values\((?:(.+),\s*)?([a-zA-Z_][a-zA-Z0-9_]*)\)\s*$/;
+
+export function migrateVariableQuery(rawQuery: string | LokiVariableQuery): LokiVariableQuery {
```

Comment:
1. I want to believe the name is intended and not random or arbitraty. I understand the "migration" part comes from the legacy string form to the `LokiVariableQuery` form.
2. Sure. I don't have enough context to have an opinion about this. Did you have a chance of seeing that I took this from here https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/datasource.ts#L314 ?

Reply:
Yep, saw that. I also don't want to say that this is something we need to do immediately. But since we did not have the LogQL parser for a long time, there are still some places where we use RegEx over parsing. Furthermore, I am unsure to which extend we want to use the parser, since it should be aligned with the parser in Loki itself.

Thanks for the explanation!